src/ceph-daemon

```diff
@@ -0,0 +1,1289 @@
+#!/usr/bin/env python3
+
+DEFAULT_IMAGE='ceph/daemon-base'
+DATA_DIR='/var/lib/ceph'
+LOG_DIR='/var/log/ceph'
+UNIT_DIR='/etc/systemd/system'
+VERSION='unknown development version'
+
+PODMAN_PREFERENCE = ['podman', 'docker']  # prefer podman to docker
+
+"""
+You can invoke ceph-daemon in two ways:
+
+1. The normal way, at the command line.
+
+2. By piping the script to the python3 binary.  In this latter case, you should
+   prepend one or more lines to the beginning of the script.
+
+   For arguments,
+
+       injected_argv = [...]
+
+   e.g.,
+
+       injected_argv = ['ls']
+
+   For reading stdin from the '--config-and-json -' argument,
+
+       injected_stdin = '...'
+"""
+
+import argparse
+import configparser
+import json
+import logging
+import os
+import subprocess
+import sys
+import tempfile
+import time
+from distutils.spawn import find_executable
+
+try:
+    from StringIO import StringIO
+except:
+    pass
+try:
+    from io import StringIO
+except:
+    pass
+
+logging.basicConfig(level=logging.INFO)
+
+podman_path = None
+
+##################################
+
+def pathify(p):
+    if '/' not in p:
+        return './' + p
+    return p
+
+def get_hostname():
+    import socket
```

Comment:
is there a specific reason to have this import inside the function? can we get them all to the top?